include/robot_interfaces/opencv_driver.hpp

```diff
@@ -0,0 +1,73 @@
+#pragma once
+
+#include <ctime>
+#include <iostream>
+
+#include <real_time_tools/process_manager.hpp>
+#include <real_time_tools/thread.hpp>
+#include <real_time_tools/threadsafe/threadsafe_timeseries.hpp>
+#include <real_time_tools/timer.hpp>
+#include "opencv2/opencv.hpp"
+
+namespace robot_interfaces
+{
+/**
+ * @brief Driver for interacting with any camera using OpenCV.
+ */
+
+template <typename OpenCVObservation>
+class OpenCVDriver
+{
+public:
+    cv::VideoCapture video_capture;
+    real_time_tools::Timer timer;
+
+    OpenCVDriver()
+    {
+        cv::VideoCapture cap(0);
+        video_capture = cap;
+    }
+
+    /**
+     * @brief Find out if the camera can be accessed and
+     * if the video capture has been started.
+     */
+
+    int is_grabbing_successful()
```

Comment:
Better return a bool value here, this would make it more obvious what values are to be expected.